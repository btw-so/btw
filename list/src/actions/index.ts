export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './list';
export * from './notes';