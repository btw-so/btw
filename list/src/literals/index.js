import { keyMirror } from "@gilbarbara/helpers";

// import { Status } from "types";

export const ActionTypes = keyMirror({
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  HIDE_ALERT: undefined,
  SHOW_ALERT: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
  GET_USER: undefined,
  GET_USER_SUCCESS: undefined,
  GET_USER_FAILURE: undefined,
  GENERATE_OTP: undefined,
  GENERATE_OTP_SUCCESS: undefined,
  GENERATE_OTP_FAILURE: undefined,
  VERIFY_OTP: undefined,
  VERIFY_OTP_SUCCESS: undefined,
  VERIFY_OTP_FAILURE: undefined,
  UPDATE_USER: undefined,
  UPDATE_USER_SUCCESS: undefined,
  UPDATE_USER_FAILURE: undefined,
  RESET_USER_STATE: undefined,
  GET_LIST: undefined,
  GET_LIST_SUCCESS: undefined,
  GET_LIST_FAILURE: undefined,
  NEW_LIST_NODE: undefined,
  EDIT_LIST_NODE_POS: undefined,
  EDIT_LIST_NODE_CONTENT: undefined,
  UPSERT_LIST_NODE: undefined,
  CHANGE_SELECTED_NODE: undefined,
  BATCH_PUSH_NODES: undefined,
  SAVE_NOTE_CONTENT: undefined,
  GET_NOTE: undefined,
  GET_NOTE_SUCCESS: undefined,
  GET_NOTE_FAILURE: undefined,
  ADD_FILE: undefined,
  ADD_FILE_SUCCESS: undefined,
  ADD_FILE_FAILURE: undefined,
  GET_FILE: undefined,
  GET_FILE_SUCCESS: undefined,
  GET_FILE_FAILURE: undefined,
  GET_PINNED_NODES: undefined,
  GET_PINNED_NODES_SUCCESS: undefined,
  GET_PINNED_NODES_FAILURE: undefined,
  GET_PUBLIC_NOTE: undefined,
  GET_PUBLIC_NOTE_SUCCESS: undefined,
  GET_PUBLIC_NOTE_FAILURE: undefined,
  SEARCH_NODES: undefined,
  SEARCH_NODES_SUCCESS: undefined,
  SEARCH_NODES_FAILURE: undefined,
});

export const STATUS = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error",
  PINNED: "pinned",
};
