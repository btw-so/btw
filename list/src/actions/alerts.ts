import React from 'react';
import { uuid } from '@gilbarbara/helpers';
import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from 'modules/helpers';

import { ActionTypes } from 'literals';

import { ShowAlertOptions } from 'types';

export const hideAlert = createAction<string>(ActionTypes.HIDE_ALERT);

export const showAlert = createAction(
  ActionTypes.SHOW_ALERT,
  (message: React.ReactNode, options: ShowAlertOptions) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return actionPayload({
      id: options.id || uuid(),
      icon: options.icon || 'dot-circle-o',
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    });
  },
);
