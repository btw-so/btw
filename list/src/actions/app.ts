import { createAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';

import { SetAppOOptions } from 'types';

export const setAppOptions = createAction<SetAppOOptions>(ActionTypes.SET_APP_OPTIONS);
