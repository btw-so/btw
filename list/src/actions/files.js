import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";


export const addFile = createAction(
  ActionTypes.ADD_FILE,
  (data) => ({
    payload: data,
  })
);
export const addFileSuccess = createAction(
  ActionTypes.ADD_FILE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const addFileFailure = createAction(
  ActionTypes.ADD_FILE_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const getFile = createAction(ActionTypes.GET_FILE, (data) => ({
  payload: data,
}));
export const getFileSuccess = createAction(
  ActionTypes.GET_FILE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getFileFailure = createAction(
  ActionTypes.GET_FILE_FAILURE,
  (data) => ({
    payload: data,
  })
);
