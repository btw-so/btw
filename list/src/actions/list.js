import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";

export const batchPushNodes = createAction(
  ActionTypes.BATCH_PUSH_NODES,
  (data) => ({
    payload: data,
  })
);

export const getList = createAction(ActionTypes.GET_LIST, (data) => ({
  payload: data,
}));
export const getListSuccess = createAction(
  ActionTypes.GET_LIST_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getListFailure = createAction(
  ActionTypes.GET_LIST_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const newListNode = createAction(ActionTypes.NEW_LIST_NODE, (data) => ({
  payload: data,
}));

export const editListNodePos = createAction(
  ActionTypes.EDIT_LIST_NODE_POS,
  (data) => ({
    payload: data,
  })
);

export const editListNodeContent = createAction(
  ActionTypes.EDIT_LIST_NODE_CONTENT,
  (data) => ({
    payload: data,
  })
);

export const upsertListNode = createAction(
  ActionTypes.UPSERT_LIST_NODE,
  (data) => ({
    payload: data,
  })
);

export const changeSelectedNode = createAction(
  ActionTypes.CHANGE_SELECTED_NODE,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodes = createAction(
  ActionTypes.GET_PINNED_NODES,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodesSuccess = createAction(
  ActionTypes.GET_PINNED_NODES_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodesFailure = createAction(
  ActionTypes.GET_PINNED_NODES_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const getPublicNote = createAction(
  ActionTypes.GET_PUBLIC_NOTE,
  (data) => ({
    payload: data,
  })
);

export const getPublicNoteSuccess = createAction(
  ActionTypes.GET_PUBLIC_NOTE_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const getPublicNoteFailure = createAction(
  ActionTypes.GET_PUBLIC_NOTE_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const searchNodes = createAction(
  ActionTypes.SEARCH_NODES,
  (data) => ({
    payload: data,
  })
);

export const searchNodesSuccess = createAction(
  ActionTypes.SEARCH_NODES_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const searchNodesFailure = createAction(
  ActionTypes.SEARCH_NODES_FAILURE,
  (data) => ({
    payload: data,
  })
);
