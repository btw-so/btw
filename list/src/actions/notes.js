import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";


export const saveNoteContent = createAction(
  ActionTypes.SAVE_NOTE_CONTENT,
  (data) => ({
    payload: data,
  })
);

export const getNote = createAction(ActionTypes.GET_NOTE, (data) => ({
  payload: data,
}));
export const getNoteSuccess = createAction(
  ActionTypes.GET_NOTE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getNoteFailure = createAction(
  ActionTypes.GET_NOTE_FAILURE,
  (data) => ({
    payload: data,
  })
);
