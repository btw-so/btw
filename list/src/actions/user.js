import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";

export const login = createAction(ActionTypes.USER_LOGIN_REQUEST);
export const loginSuccess = createAction(ActionTypes.USER_LOGIN_SUCCESS);

export const logOut = createAction(ActionTypes.USER_LOGOUT_REQUEST);
export const logOutSuccess = createAction(ActionTypes.USER_LOGOUT_SUCCESS);

export const getUser = createAction(ActionTypes.GET_USER);
export const getUserSuccess = createAction(
  ActionTypes.GET_USER_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getUserFailure = createAction(
  ActionTypes.GET_USER_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const generateOtp = createAction(ActionTypes.GENERATE_OTP, (data) => ({
  payload: data,
}));
export const generateOtpSuccess = createAction(
  ActionTypes.GENERATE_OTP_SUCCESS,
  ({ success, error, data }) => ({ payload: { success, error, data } })
);
export const generateOtpFailure = createAction(
  ActionTypes.GENERATE_OTP_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const verifyOtp = createAction(
  ActionTypes.VERIFY_OTP,
  ({ email, otp } = {}) => ({
    payload: { email, otp },
  })
);
export const verifyOtpSuccess = createAction(
  ActionTypes.VERIFY_OTP_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const verifyOtpFailure = createAction(
  ActionTypes.VERIFY_OTP_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const updateUser = createAction(
  ActionTypes.UPDATE_USER,
  ({ name, slug, bio, pic, twitter, linkedin, instagram, settings } = {}) => ({
    payload: { name, slug, bio, pic, twitter, linkedin, instagram, settings },
  })
);
export const updateUserSuccess = createAction(
  ActionTypes.UPDATE_USER_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const updateUserFailure = createAction(
  ActionTypes.UPDATE_USER_FAILURE,
  (data) => ({
    payload: data,
  })
);
export const resetState = createAction(
  ActionTypes.RESET_USER_STATE,
  (data) => ({
    payload: data,
  })
);
