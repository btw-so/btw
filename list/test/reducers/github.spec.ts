import { ActionTypes } from 'literals';

import { getRepos } from 'actions';
import reducer from 'reducers/github';

import { emptyAction } from 'test-utils';

describe('Github', () => {
  it('should return the initial state', () => {
    expect(reducer.github(undefined, emptyAction)).toMatchSnapshot();
  });

  describe('GITHUB_GET_REPOS', () => {
    it('should handle REQUEST', () => {
      expect(reducer.github(undefined, getRepos('react'))).toMatchSnapshot();
    });

    it('should handle SUCCESS', () => {
      const initialState = reducer.github(undefined, getRepos('react'));

      expect(
        reducer.github(initialState, {
          type: ActionTypes.GITHUB_GET_REPOS_SUCCESS,
          payload: {
            data: [{ name: 'one' }],
          },
          meta: {
            query: 'react',
            updatedAt: 1234567890,
          },
        }),
      ).toMatchSnapshot();
    });

    it('should handle FAILURE', () => {
      const initialState = reducer.github(undefined, getRepos('react'));

      expect(
        reducer.github(initialState, {
          type: ActionTypes.GITHUB_GET_REPOS_FAILURE,
          payload: {
            message: 'Something failed',
          },
          meta: {
            query: 'react',
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
