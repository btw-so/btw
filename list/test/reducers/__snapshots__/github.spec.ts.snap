// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github GITHUB_GET_REPOS should handle FAILURE 1`] = `
Object {
  "topics": Object {
    "react": Object {
      "cached": false,
      "data": Array [],
      "message": Object {
        "message": "Something failed",
      },
      "status": "error",
      "updatedAt": 0,
    },
  },
}
`;

exports[`Github GITHUB_GET_REPOS should handle REQUEST 1`] = `
Object {
  "topics": Object {
    "react": Object {
      "cached": false,
      "data": Array [],
      "message": "",
      "status": "running",
      "updatedAt": 0,
    },
  },
}
`;

exports[`Github GITHUB_GET_REPOS should handle SUCCESS 1`] = `
Object {
  "topics": Object {
    "react": Object {
      "cached": false,
      "data": Object {
        "data": Array [
          Object {
            "name": "one",
          },
        ],
      },
      "message": "",
      "status": "success",
      "updatedAt": 1234567890,
    },
  },
}
`;

exports[`Github should return the initial state 1`] = `
Object {
  "topics": Object {},
}
`;
